apps/target_manager.py

```diff
@@ -40,15 +49,18 @@ def create_target(targets_json, version, compose_apps, ota_lite_tag, git_sha):
             target['custom']['version'] = version
             if tag:
                 target['custom']['tags'] = [tag]
-            apps = {}
             target['custom']['containers-sha'] = git_sha
             if compose_apps:
                 target['custom']['docker_compose_apps'] = compose_apps
             elif 'docker_compose_apps' in target['custom']:
                 del target['custom']['docker_compose_apps']
             if 'docker_apps' in target['custom']:
                 del target['custom']['docker_apps']
-            logging.info('Targets with apps: %r', target)
+            logging.info('New Target with {}\n{}'.
+                         format(tgt_name, json.dumps(target, ensure_ascii=True, indent=2)))
+            new_target_counter += 1
 
     with open(targets_json, 'w') as f:
         json.dump(data, f, indent=2)
+
+    return new_target_counter > 0
```

Comment:
can you articulate the changes here a bit. Some new things just look like changes to add more logging. But there are some other changes as well. I want to make sure I'm not missing something subtle

Reply:
New things:
1. The build number becomes optional. It's useful for testing purposes when you run it out of the scope of any CI/JobServ job. In this case, the build number (aka version) is calculated as the latest version found in the current targets.json + 1.

2. It returns a list of the new Targets that just been created. The follow-up "preloading" procedure needs a list of new Targets as an input so it knows what apps to dump/preload (apps' images arch/platform).